File: sdk/php/src/SdkConfig.php
<?php

declare(strict_types=1);

namespace ExperimentationPlatform;

/**
 * Configuration for the ExperimentationClient SDK.
 *
 * Immutable value object holding all connection and caching parameters.
 */
class SdkConfig
{
    /**
     * @param string $baseUrl       Base URL of the Experimentation Platform API (e.g. https://api.example.com)
     * @param string $apiKey        API key for authenticating SDK requests
     * @param int    $cacheTtl      Cache time-to-live in seconds (default 300 = 5 minutes)
     * @param int    $timeout       HTTP request timeout in seconds (default 10)
     * @param int    $maxCacheSize  Maximum number of entries in the in-memory cache (default 1000)
     *
     * @throws \InvalidArgumentException if baseUrl or apiKey are empty
     */
    public function __construct(
        public readonly string $baseUrl,
        public readonly string $apiKey,
        public readonly int $cacheTtl = 300,
        public readonly int $timeout = 10,
        public readonly int $maxCacheSize = 1000,
    ) {
        if (empty($baseUrl)) {
            throw new \InvalidArgumentException('baseUrl is required');
        }
        if (empty($apiKey)) {
            throw new \InvalidArgumentException('apiKey is required');
        }
        if ($cacheTtl < 0) {
            throw new \InvalidArgumentException('cacheTtl must be >= 0');
        }
        if ($timeout <= 0) {
            throw new \InvalidArgumentException('timeout must be > 0');
        }
        if ($maxCacheSize <= 0) {
            throw new \InvalidArgumentException('maxCacheSize must be > 0');
        }
    }
}

File: sdk/dotnet/src/ExperimentationPlatform/SdkConfig.cs
namespace ExperimentationPlatform;

/// <summary>
/// Configuration for the ExperimentationClient SDK.
/// </summary>
public class SdkConfig
{
    /// <summary>The base URL of the Experimentation Platform API (e.g. "https://api.example.com").</summary>
    public string BaseUrl { get; init; }

    /// <summary>The API key used for authentication.</summary>
    public string ApiKey { get; init; }

    /// <summary>How long to cache feature flag responses in seconds. Default: 300 (5 minutes).</summary>
    public int CacheTtlSeconds { get; init; } = 300;

    /// <summary>HTTP request timeout in seconds. Default: 10.</summary>
    public int TimeoutSeconds { get; init; } = 10;

    /// <summary>Maximum number of entries to hold in the in-memory cache. Default: 1000.</summary>
    public int MaxCacheSize { get; init; } = 1000;

    /// <summary>
    /// Initialises a new SdkConfig.
    /// </summary>
    /// <param name="baseUrl">Required. The API base URL (must not be null or empty).</param>
    /// <param name="apiKey">Required. The API key (must not be null or empty).</param>
    /// <exception cref="ArgumentException">Thrown if baseUrl or apiKey is null or empty.</exception>
    public SdkConfig(string baseUrl, string apiKey)
    {
        if (string.IsNullOrWhiteSpace(baseUrl))
            throw new ArgumentException("BaseUrl is required and must not be empty.", nameof(baseUrl));
        if (string.IsNullOrWhiteSpace(apiKey))
            throw new ArgumentException("ApiKey is required and must not be empty.", nameof(apiKey));

        BaseUrl = baseUrl.TrimEnd('/');
        ApiKey = apiKey;
    }
}
